src/core/users/service/users.service.ts

```diff
@@ -1,25 +1,69 @@
-import { Injectable, NotFoundException } from "@nestjs/common";
+import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
 import { CreateUserDto } from "../dto/create-user.dto";
 import { UpdateUserDto } from "../dto/update-user.dto";
 import { InjectRepository } from "@nestjs/typeorm";
 import { User } from "../entities/user.entity";
 import { Repository } from "typeorm";
 import * as bcryptjs from "bcryptjs";
+import { PageDto } from "@src/common/dto/page.dto";
+import { PageOptionsDto } from "@src/common/dto/pageOptions.dto";
+import { PageMetaDto } from "@src/common/dto/page.meta.dto";
+import { DptosService } from "@src/core/dptos/services/dptos.service";
 
 @Injectable()
 export class UsersService {
   constructor(
     @InjectRepository(User)
     private readonly userRepository: Repository<User>,
+    private readonly dptoService: DptosService,
   ) {}
 
   async create(createUserDto: CreateUserDto): Promise<User> {
+    const user = await this.findByEmailWithPassword(createUserDto.email);
+
+    const dpto = await this.dptoService.findOne(createUserDto.departmentId);
+
+    const userDpto = await this.userRepository.find({
+      relations: ["department"],
+      where: {
+        department: {
+          id: dpto.id,
+        },
+      },
+    });
+
+    if (user) {
+      throw new BadRequestException("El usuario ya existe");
+    }
+
+    if (!dpto) {
+      throw new BadRequestException("El departamento no existe");
+    }
+
+    if (userDpto && userDpto.some((user) => user.role === "dpto")) {
```

Comment:
Usen enums
```suggestion
    if (userDpto && userDpto.some((user) => user.role === UseRole.Dpto)) {
```

Pero... ¿porque se llama "dpto" el rol? seria mejor usuario u algo asi, pero bueno, x, dejemoslo asi